sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/spice_indexer.py

```diff
@@ -0,0 +1,386 @@
+"""Functions to write SPICE ingested files to EFS."""
+
+import logging
+import os
+from datetime import datetime
+from pathlib import Path
+
+import boto3
+import spiceypy
+from imap_data_access import SPICEFilePath
+from sqlalchemy.dialects.postgresql import insert
+
+from ..database import database as db
+from ..database import models
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+# Define constants needed in the file
+SPACECRAFT_ID = -43
+minimum_mission_time = datetime(2023, 1, 1)
+maximum_mission_time = datetime(2145, 1, 1)
+MAXIMUM_DATETIME_INTERVAL = [[minimum_mission_time, maximum_mission_time]]
+MAXIMUM_SCLK_INTERVAL = [
+    ["1/0410227203:00000", "1/4288750963:38093"]
+]  # Calculated from the above datetimes seperately
+MAXIMUM_J2000_INTERVAL = [
+    [725803269.1839136, 4575787269.183866]
+]  # Calculated from the above datetimes seperately
+
+# Set constants for the time interval calculations
+COVERAGE_ANGULAR_VELOCITY_ONLY = False  # Only include segments with angular velocity?
+COVERAGE_SPICE_ARRAY_LENGTH = 10000  # Use an array size of 10000 for coverage calc
+COVERAGE_LEVEL = "INTERVAL"  # the granularity at which the coverage is examined
+COVERAGE_TOLERANCE = 1000000.0  # tolerance value expressed in ticks of the spacecraft
```

Comment:
Is this roughly 20 seconds? Assuming 20 microseconds / tick which I think I've seen elsewhere. I have no idea what a reasonable number is, but as an FYI that is longer than a spin which is roughly 15-seconds if that matters.